### PR TITLE
adds the parsing of indexes from the DB to the worksets

### PR DESCRIPTION
**Pull request by Galithil**
Catches the indexes of the samples for the workset DB

### CODE REVIEW

File: LIMS2DB/classes.py [hunk 1]
```diff
@@ -297,6 +316,20 @@ def build(self):
...
+                #Grabbing indexes
+                # Get all artifacts for given sample
+                query = "select art.* from artifact art \
+                    inner join artifact_sample_map asm on asm.artifactid=art.artifactid \
+                    inner join sample sa on sa.processid=asm.processid \
+                    where sa.processid = {sapid};".format(sapid=sample.processid)
+                try:
+                    artifacts = self.session.query(Artifact).from_statement(text(query)).all()
+                    for art in artifacts:
+                            if art.reagentlabels is not None and len(art.reagentlabels) == 1:
```
Comment (vezzi) [inline LIMS2DB/classes.py]: seems to be extra spaces here

Comment (vezzi) [inline LIMS2DB/classes.py]: what happens if the length is larger than 1? well nothing ... but are there cases where art.reagnetlabels is larger than 1?

Comment (Galithil) [inline LIMS2DB/classes.py]: Yes, and I want to avoid them, because it means that the artifact is a pool and has potentially a billion of reagent labels.

Comment (vezzi) [inline LIMS2DB/classes.py]: ok makes perfect sense. Can you add a comment for future "brave" programmers that will put their hands on this code? thanks

Comment (Galithil) [inline LIMS2DB/classes.py]: handled in https://github.com/SciLifeLab/LIMS2DB/pull/48/commits/d2aa88bb2708c5cb0cdf8c0ca403c9247d9d2ff0

File: LIMS2DB/classes.py [hunk 2]
```diff
@@ -203,6 +203,25 @@ def __init__(self, session, log, step):
...
+        if "NoIndex" in chain:
+            return chain
+        if '(' not in chain:
+            barcode = chain
+        else:
+            pattern = re.compile("\(([A-Z\-]+)\)")
+            matches = pattern.search(chain)
+            if matches.group(1):
+                barcode = matches.group(1)
+        matches = bcp.match(barcode)
```
Comment (Galithil) [inline LIMS2DB/classes.py]: handled in https://github.com/SciLifeLab/LIMS2DB/pull/48/commits/d2aa88bb2708c5cb0cdf8c0ca403c9247d9d2ff0

**Pull request merged**